rm-datasource/src/main/java/org/apache/seata/rm/datasource/DataSourceProxy.java

```diff
@@ -399,27 +403,36 @@ public String getKernelVersion() {
         return kernelVersion;
     }
 
+    public String getVariableValue(String name) {
+        return variables.get(name);
+    }
+
     private void validMySQLVersion(Connection connection) {
         if (!JdbcConstants.MYSQL.equals(dbType)) {
             return;
         }
-        try (PreparedStatement preparedStatement = connection.prepareStatement("SELECT VERSION()");
-             ResultSet versionResult = preparedStatement.executeQuery()) {
-            if (versionResult.next()) {
-                String version = versionResult.getString("VERSION()");
-                if (StringUtils.isBlank(version)) {
-                    return;
-                }
-                int dashIdx = version.indexOf('-');
-                // in mysql: 5.6.45, in polardb-x: 5.6.45-TDDL-xxx
-                if (dashIdx > 0) {
-                    kernelVersion = version.substring(0, dashIdx);
-                    productVersion = version.substring(dashIdx + 1);
-                } else {
-                    kernelVersion = version;
-                    productVersion = version;
+        try (PreparedStatement preparedStatement = connection.prepareStatement("SHOW VARIABLES");
```

Comment:
是不是应该弄一个variables的元数据存储的类，否则都放在datasourceproxy里，有点臃肿
Should I get a metadata storage class for variables, otherwise it will be put in datasourceproxy, which is a bit bloated?